runtime/include/tt/runtime/runtime.h

```diff
@@ -55,26 +56,48 @@ createTensor(std::vector<std::shared_ptr<void>> &data, TensorDesc const &desc,
                       desc.dataType, strategy);
 }
 
-tt::target::DataType getTensorDataType(Tensor tensor);
+tt::target::DataType getTensorDataType(const Tensor &tensor);
 
 size_t getNumAvailableDevices();
 
 Device openDevice(DeviceIds const &deviceIds, size_t numHWCQs = 1);
 
 void closeDevice(Device device);
 
-Event submit(Device device, Binary executable, std::uint32_t programIndex,
-             std::vector<Tensor> const &inputs,
-             std::vector<Tensor> const &outputs);
-
 void wait(Event event);
 
+Tensor toHost(const Tensor &tensor, bool untilize = false);
+
+Tensor toDevice(const Tensor &tensor, Device &device);
+
+Tensor toDevice(const Tensor &tensor, Device &device, const Layout &layout);
+
+Tensor toLayout(const Tensor &tensor, const Layout &layout);
+
+Layout getLayout(const Binary &executableHandle, std::uint32_t programIndex,
+                 std::uint32_t inputIndex);
+
 std::string getOpDebugString(OpContext opContextHandle);
 
 Tensor getOpOutputTensor(OpContext opContextHandle,
                          CallbackContext programContextHandle);
 
-std::vector<float> getTensorData(Tensor tensor);
+std::vector<float> getTensorData(const Tensor &tensor);
+
+void memcpy(void *dst, const Tensor &src);
+
+void memcpy(Tensor &dst, const Tensor &src);
+
+void deallocateTensor(Tensor &tensor, bool force = false);
+
+std::map<std::uint32_t, Tensor> submit(Device &deviceHandle,
```

Comment:
I'm actually not sure we should return a `std::map`, I think a `std::vector` will suffice.  The `global_id` is only global to the flatbuffer binary internally, so there's nothing that the FE could actually do to make sense of this id.

A vector is better because then the FE can use the output index as an ID to correlate tensors between programs.